src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/CommandProcessor.java

```diff
@@ -1773,20 +1773,44 @@ public void doit(Tokens t) {
                 sysProps.run();
             }
         },
-        new Command("dumpheap", "dumpheap [filename]", false) {
+        new Command("dumpheap", "dumpheap [gz=<1-9>] [filename]", false) {
             public void doit(Tokens t) {
-                if (t.countTokens() > 1) {
+                if (t.countTokens() > 2) {
                     usage();
                 } else {
                     JMap jmap = new JMap();
-                    String filename;
-                    if (t.countTokens() == 1) {
-                        filename = t.nextToken();
+                    String filename = "heap.bin";
+                    int gzlevel = 0;
+                    /* Parse "gz=" option. */
+                    String option = t.nextToken();
+                    String[] keyValue = option.split("=");
+                    if (keyValue[0].equals("gz")) {
+                        String level = keyValue[1];
```

Comment:
If the user specified `gz` option incorrectly (e.g. `dumpheap gz` : without the level), `ArrayIndexOutOfBoundsException` will be thrown.